source/brailleDisplayDrivers/seikantk.py

```diff
@@ -57,66 +59,127 @@
+
+
+def isSeikaBluetoothName(bluetoothName: str) -> bool:
+	return bool(seikaBluetoothNameRegex.match(bluetoothName))
+
+
+def isSeikaBluetoothDeviceInfo(deviceInfo: typing.Dict[str, str]) -> bool:
+	# bluetoothName is listed in information from L{hwPortUtils.listComPorts} when 'hwIo' debug logging
+	# category is enabled.
+	btNameKey = "bluetoothName"
+	return (
+		btNameKey in deviceInfo
+		and isSeikaBluetoothName(deviceInfo["bluetoothName"])
+	)
+
+
+def isSeikaBluetoothDeviceMatch(match: DeviceMatch) -> bool:
+	return isSeikaBluetoothDeviceInfo(match.deviceInfo)
+
 
 class BrailleDisplayDriver(braille.BrailleDisplayDriver):
 	_dev: hwIo.IoBase
 	name = SEIKA_NAME
 	# Translators: Name of a braille display.
 	description = _("Seika Notetaker")
-	path = ""
 	isThreadSafe = True
-	for d in hwPortUtils.listHidDevices():
-		if d["hardwareID"].startswith(hidvidpid):
-			path = d["devicePath"]
 
 	@classmethod
 	def getManualPorts(cls) -> typing.Iterator[typing.Tuple[str, str]]:
 		"""@return: An iterator containing the name and description for each port.
 		"""
-		return braille.getSerialPorts()
+		return braille.getSerialPorts(isSeikaBluetoothDeviceInfo)
 
-	def __init__(self, port="hid"):
+	def __init__(self, port: typing.Union[None, str, DeviceMatch]):
 		super().__init__()
 		self.numCells = 0
 		self.numBtns = 0
 		self.numRoutingKeys = 0
 		self.handle = None
-
 		self._hidBuffer = b""
 		self._command: typing.Optional[bytes] = None
 		self._argsLen: typing.Optional[int] = None
-		log.info(f"Seika Notetaker braille driver path: {self.path}")
-
-		if self.path == "":
-			raise RuntimeError("No MINI-SEIKA display found, no path found")
-		self._dev = dev = hwIo.Hid(path=self.path, onReceive=self._onReceive)
-		if dev._file == INVALID_HANDLE_VALUE:
-			raise RuntimeError("No MINI-SEIKA display found, open error")
-		dev.setFeature(SEIKA_CONFIG)  # baud rate, stop bit usw
-		dev.setFeature(SEIKA_CMD_ON)  # device on
+
+		log.debug(f"Seika Notetaker braille driver: ({port!r})")
+		dev: typing.Optional[typing.Union[hwIo.Hid, hwIo.Serial]] = None
+		for match in self._getTryPorts(port):
+			self.isHid = match.type == bdDetect.KEY_HID
+			self.isSerial = match.type == bdDetect.KEY_SERIAL
+			try:
+				if self.isHid:
+					log.info(f"Trying Seika notetaker on USB-HID")
+					self._dev = dev = hwIo.Hid(
+						path=match.port,  # for a Hid match type 'port' is actually 'path'.
+						onReceive=self._onReceiveHID
+					)
+					dev.setFeature(SEIKA_HID_FEATURES)  # baud rate, stop bit usw
+					dev.setFeature(SEIKA_CMD_ON)  # device on
+				elif self.isSerial:
+					log.info(f"Trying Seika notetaker on Bluetooth (serial) port:{match.port}")
+					self._dev = dev = hwIo.Serial(
+						port=match.port,
+						onReceive=self._onReceiveSerial,
+						baudrate=BAUD,
+						parity=serial.PARITY_NONE,
+						bytesize=serial.EIGHTBITS,
+						stopbits=serial.STOPBITS_ONE,
+					)
+					# Does the device need to be sent SEIKA_CMD_ON as per USB-HID?
```

Comment:
I think the answer is that it isn't required, I'd like to leave some comment here to highlight the difference between serial and HID.  But I'll convert it to a statement rather than a question.